### PR TITLE
Hello world app

### PR DESCRIPTION
**Pull request by BenRamchandani**
Add a ktor app, this is mostly the default app generated by IntelliJ with a few plugins added and

* An SSL listener with a self-signed cert
* Docker build (ktor has a built in `buildImage` gradle task, but I figured we'd want to control it ourselves)
* GitHub action for building and running tests

I've used gradle as that's what it came with, does anyone have a preference there?

I haven't tried deploying it yet, I'll do that today.

### CODE REVIEW

File: auth-service/gradle.properties [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ktor_version=2.3.1
```
Comment (hugh-emerson) [inline auth-service/gradle.properties:1]: 2.3.2 released on Wednesday :P

File: auth-service/src/main/kotlin/uk/gov/communities/delta/auth/SelfSignedSSLCertKeystore.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package uk.gov.communities.delta.auth
+
+import io.ktor.network.tls.certificates.*
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.security.KeyStore
+
+/*
+* We deploy behind an AWS ALB which doesn't validate SSL certificates anyway
+* so generate a self-signed certificate on startup
```
[the remark this answers is not in the document]
Comment (BenRamchandani) [reply auth-service/src/main/kotlin/uk/gov/communities/delta/auth/SelfSignedSSLCertKeystore.kt:10]: Pretty minimal, it offers some protection against a passive eavesdropper that can see VPC traffic. I was planning to let other apps continue connecting to the ALB like they are at the moment, which does have a valid cert.

File: auth-service/src/main/kotlin/uk/gov/communities/delta/auth/Application.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import io.ktor.server.application.*
+import io.ktor.server.engine.*
+import io.ktor.server.netty.*
+import uk.gov.communities.delta.auth.plugins.configureMonitoring
+import uk.gov.communities.delta.auth.plugins.configureRouting
+import uk.gov.communities.delta.auth.plugins.configureSecurity
+import uk.gov.communities.delta.auth.plugins.configureSerialization
+
+fun main() {
```
Comment (hugh-emerson) [inline auth-service/src/main/kotlin/uk/gov/communities/delta/auth/Application.kt:11]: What did you use to set up the project? Intellij's new ktor project?

File: auth-service/src/main/kotlin/uk/gov/communities/delta/auth/plugins/Routing.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package uk.gov.communities.delta.auth.plugins
+
+import io.ktor.server.routing.*
+import io.ktor.server.response.*
+import io.ktor.server.application.*
+
+fun Application.configureRouting() {
```
Comment (hugh-emerson) [inline auth-service/src/main/kotlin/uk/gov/communities/delta/auth/plugins/Routing.kt:7]: How come there was no need to `install(Routing)`? From my very brief look at ktor docs it says to do that.

**Review by hugh-emerson — APPROVED** [review]
As part of setting up a skeleton app, we could also look into these now but fine by me if you're doing them as part of DT-481
* environment specific conf files
* Dependency injection (koin?)
* Project structure, like an example of a route defined in its own file because we aren't going to want to put everything in Routing.kt.

File: auth-service/src/main/kotlin/uk/gov/communities/delta/auth/plugins/Routing.kt (hunk 4 above)
Comment (BenRamchandani) [reply auth-service/src/main/kotlin/uk/gov/communities/delta/auth/plugins/Routing.kt:7]: Looks like have a `routing {...}` block does it for you

https://ktor.io/docs/routing-in-ktor.html
> there is a convenient `routing` function that makes it simpler to install routing
> [...] `install(Routing)` is replaced with the `routing` function

File: auth-service/src/main/kotlin/uk/gov/communities/delta/auth/Application.kt (hunk 3 above)
Comment (BenRamchandani) [reply auth-service/src/main/kotlin/uk/gov/communities/delta/auth/Application.kt:11]: Yep

**Comment by hugh-emerson** [comment]
Also, yeah I'm happy with gradle. I prefer it to maven and some of the other projects already use gradle.

File: auth-service/gradle.properties (hunk 1 above)
Comment (BenRamchandani) [reply auth-service/gradle.properties]: Updated, why not :D

**Comment by BenRamchandani** [comment]
Will address the other points in DT-481

**Pull request merged**